rest/data/project_test.go

```diff
@@ -637,31 +638,63 @@ func TestHideBranch(t *testing.T) {
 	}
 	require.NoError(t, vars.Insert(t.Context()))
 
-	err := HideBranch(t.Context(), project.Id)
-	assert.NoError(t, err)
+	// Insert GitHub app and ensure it inserts a fake parameter.
+	var ghAppID int64 = 90
+	ghPrivateKey := []byte("secret")
+	require.NoError(t, project.SetGithubAppCredentials(t.Context(), ghAppID, ghPrivateKey))
 
-	hiddenProj, err := model.FindMergedProjectRef(t.Context(), project.Id, "", true)
-	assert.NoError(t, err)
-	skeletonProj := model.ProjectRef{
-		Id:        project.Id,
-		Owner:     repo.Owner,
-		Repo:      repo.Repo,
-		Branch:    project.Branch,
-		RepoRefId: repo.Id,
-		Enabled:   false,
-		Hidden:    utility.TruePtr(),
-	}
-	assert.Equal(t, skeletonProj, *hiddenProj)
```

Comment:
I separated out all the checks to their own subtest for clarity, since we are checking for a couple of different things now.

Reply:
Bless 🙏